### ISSUE TITLE
Stage action views can overwrite in slow environments

### ISSUE THREAD

**Issue by gsteimer** [tsgrp/hpi#537]
See the screencam below.  Here's what I'm seeing:

1. Load an action in the stage.  In the screencam, I have the view all documents action launching automatically upon folder load (this is very common in the insurance scenario).
1. While the first action is waiting to fully load, click on another action that:
  * displays using the same action handler as the first action
  * will display quickly.
1. The second action displays
1. The first action returns, and takes over a portion of the view causing a weird view frankenstein

![stage-action-view-overwrite](https://cloud.githubusercontent.com/assets/1377228/5760026/80f745fa-9c97-11e4-8b77-37177468f011.gif)

I'm guessing we should check before just blindly writing the view to the page.  Could this be easily fixed by targeting specific ID's vs. a common ID or class in the stage?

@zkeller @joehof @dgrumieaux @mbowen000 - thoughts?

**Comment by zkeller** [comment]
If I remember correctly, the header of the action and the actual body are rendered separately. So what is happening is:

1. View All Documents title renders
2. Add Documents is clicked
3. Add Documents View and title render
4. View All Documents View renders, replacing the Add Documents View.

This is how you end up with the wrong title and the 'right' view.

What would we want to have happen in this scenario? I assume we would want to see Add Documents.  As for the solution, I would guess we would want to more effectively kill off any view (keep it from rendering) in the docviewer when we click on an action, but I don't know off the top of my head how to do that.

**Comment by gsteimer** [comment]
Yeah, the "correct" view to display here is the second view, since that's the action that the user clicked on last.  I have a feeling like this is mostly going to be an issue for clients that have an action automatically kick off when viewing a folder.  If the user knows they want to come in an see something else, they could easily cause this scenario.

@dhartman25 - is this going to be an issue for your client?

**Comment by dhartman25** [comment]
@gsteimer  Yes - View All Documents launches automatically so I can see this easily becoming an issue.
